### ISSUE TITLE
`github.actions.workflows.name` triggers false positives since commit 35314ce23f7676b95fae18f73f9577aff2f0da03 (`some ... in` used for iterations)

### ISSUE THREAD

**Issue by iamleot** [iamleot/conftest-policies#29]
Since commit 35314ce23f7676b95fae18f73f9577aff2f0da03 a regression was likely introduced and when analyzing properly written GitHub Actions workflows `github.actions.workflows.name` is triggered.

A possible reproducer is against the `.github/workflows` of this repository at commit 35314ce23f7676b95fae18f73f9577aff2f0da03:

```console
$ conftest test --all-namespaces --policy ./policy/github .github/workflows
FAIL - .github/workflows/conftest.yml - github.actions.workflows.name - Job `{"name": "Check and validate Conftest Rego policies", "runs-on": "ubuntu-latest", "steps": [{"name": "Checkout code", "uses": "actions/checkout@v3"}, {"name": "Install conftest", "run": "mkdir -p \"${HOME}/.local/bin\"\ncurl -sL \"https://github.com/open-policy-agent/conftest/releases/download/v${CONFTEST_VERSION}/conftest_${CONFTEST_VERSION}_Linux_x86_64.tar.gz\" | tar xzf - -C \"${HOME}/.local/bin\" conftest\n"}, {"name": "Checking Rego formatting style", "run": "make check-fmt\n"}, {"name": "Unit testing conftest policies", "run": "make test\n"}]}` should have a `name` key
FAIL - .github/workflows/mdformat.yml - github.actions.workflows.name - Job `{"name": "Check Markdown formatting via mdformat", "runs-on": "ubuntu-latest", "steps": [{"name": "Checkout code", "uses": "actions/checkout@v3"}, {"name": "Install mdformat", "run": "pip install mdformat\n"}, {"name": "Check Markdown formatting via mdformat", "run": "mdformat --check .\n"}]}` should have a `name` key
FAIL - .github/workflows/actionlint.yml - github.actions.workflows.name - Job `{"name": "Check GitHub Actions workflows", "runs-on": "ubuntu-latest", "steps": [{"name": "Checkout code", "uses": "actions/checkout@v3"}, {"name": "Check GitHub Actions workflows", "uses": "docker://rhysd/actionlint:latest", "with": {"args": "-color"}}]}` should have a `name` key
FAIL - .github/workflows/conftest-self.yml - github.actions.workflows.name - Job `{"name": "Validate via personal conftest policies", "runs-on": "ubuntu-latest", "steps": [{"name": "Checkout code", "uses": "actions/checkout@v3"}, {"env": {"CONFTEST_POLICIES": "git::https://github.com/iamleot/conftest-policies.git//policy/github"}, "name": "Validate via personal conftest policies", "uses": "docker://openpolicyagent/conftest:latest", "with": {"args": "test --all-namespaces --update \"${{ env.CONFTEST_POLICIES }}\" .github/workflows"}}]}` should have a `name` key

48 tests, 44 passed, 0 warnings, 4 failures, 0 exceptions
```

Write testcase(s) to trigger that issue and fix it.

**Comment by iamleot** [comment]
After double-checking documentation regarding [in](https://www.openpolicyagent.org/docs/latest/policy-language/#futurekeywordsin) and [some](https://www.openpolicyagent.org/docs/latest/policy-language/#some-keyword) and in particular [Membership and iteration: in](https://www.openpolicyagent.org/docs/latest/policy-language/#membership-and-iteration-in) section...

- when on the left side of `in` in `some ... in` there is only an argument we eventually unify with the object
- when on the left side of `in` in `some ... in` there are two arguments we eventually unify with object and key respectively

So instead of writing:

```rego
some
```

TODO

**Comment by iamleot** [comment]
Fixed via 28bc8df.
